### PR TITLE
Upload sonobuoy's test result to s3 from cis, apps test

### PR DESCRIPTION
**Pull request by tomahawk28**
From `/apps test` I found out tekton haven't managed to upload sonobuoy result into AWS S3 if it received failures from sonobuoy test. 
image

### CODE REVIEW

File: tekton/tasks/manage-sonobuoy-results.yaml [hunk 1]
```diff
@@ -39,10 +39,6 @@ spec:
         /sonobuoy results "$outfile"
         /sonobuoy results "$outfile" --mode detailed
         mv "$outfile" "$(params.results-path)"
-
-        if /sonobuoy status --namespace "$(params.sonobuoy-namespace)" --kubeconfig "$(params.kubeconfig-path)" | grep failed >/dev/null; then
-          exit 1
```
Comment (tomahawk28) [inline tekton/tasks/manage-sonobuoy-results.yaml:44]: if we return the exit code here, tekton will not run `manage-sonobuoy-result` task afterward.

**Comment by tomahawk28** [comment]
sorry guys if I ping you too often :( but can you review this? thanks!

File: tekton/tasks/manage-sonobuoy-results.yaml (hunk 1 above)
Comment (MarcelMue) [reply tekton/tasks/manage-sonobuoy-results.yaml:44]: I checked this yesterday but I don't think this helps. We added this, so failed tests get reported as failed. I need to check the `manage-sonobuoy-result` task to see if that's still accurate.

Comment (MarcelMue) [reply tekton/tasks/manage-sonobuoy-results.yaml:44]: Yes - it's like I remembered I think. So far we only uploaded the results on success IIRC - @fiunchinho worked on the sonobuoy tasks for azure.

What information are you missing about the failed test?

Comment (MarcelMue) [reply tekton/tasks/manage-sonobuoy-results.yaml:44]: If you always want to upload to s3, you should use `finally` in your pipeline. See the `azure` pipeline for reference:

https://github.com/giantswarm/test-infra/blob/7f515f97f5e57a7c41e28b2b491330f56bebbd0d/tekton/pipelines/azure.yaml#L76

Comment (tomahawk28) [reply tekton/tasks/manage-sonobuoy-results.yaml:44]: I copied these from the `cis` task, I would fix it there, too. Thanks!

Comment (fiunchinho) [reply tekton/tasks/manage-sonobuoy-results.yaml:44]: Use [the `azure` pipeline as reference](https://github.com/giantswarm/test-infra/blob/master/tekton/pipelines/azure.yaml), as that's the one we changed more recently (I think). In that pipeline you can see the use of `finally`

**Review by fiunchinho — CHANGES_REQUESTED** [review]
We want to mark the task as failed when tests fail, so we must keep the `exit 1`. Please, use `finally` to upload the reports like in the azure pipeline.

File: tekton/pipelines/cis.yaml [hunk 2]
```diff
@@ -63,20 +63,23 @@ spec:
...
+        operator: in
+        values: ["Failed"]
     taskRef:
       name: upload-to-s3-bucket
     params:
-    - name: prow-job-id
-      value: "$(params.PROW_JOB_ID)"
-    - name: file-to-upload
-      value: $(workspaces.cluster.path)/sonobuoy-results.tar
+      - name: prow-job-id
```
Comment (tomahawk28) [inline tekton/pipelines/cis.yaml:75]: I don't know why there are differences between spacing but I would follow the `azure.yaml` as the same.

File: tekton/tasks/azure.yaml [hunk 3]
```diff
@@ -15,7 +15,6 @@ spec:
     type: string
     description: Whether or not test deletion of clusters.
     default: "1"
-    type: string
```
Comment (tomahawk28) [inline tekton/tasks/azure.yaml:18]: This was a duplicate.

File: tekton/tasks/apps.yaml [hunk 4]
```diff
@@ -26,7 +26,7 @@ spec:
             --kubeconfig "${KUBECONFIG_PATH}" \
             --namespace "$(cat "${CLUSTER_ID_PATH}")-sonobuoy" \
             --plugin $(workspaces.cluster.path)/giantswarm-plugin.yaml \
-            --plugin-env giantswarm.CP_KUBECONFIG="$(cat "${KUBECONFIG_PATH}")" \
+            --plugin-env giantswarm.CP_KUBECONFIG="$(cat /etc/kubeconfig/$(cat $(workspaces.cluster.path)/provider)) \
```
Comment (tomahawk28) [inline tekton/tasks/apps.yaml:29]: @fiunchinho I can't find a more elegant solution around this because I need to know the provider type as well. WDYT?

**Review by MarcelMue — APPROVED** [review]
LGTM if tested

**Review by fiunchinho — APPROVED** [review]
Sounds good!

**Pull request merged**